SoftLayer/managers/vs.py

```diff
@@ -1404,3 +1404,45 @@ def get_hardware_guests(self):
         object_filter = {"hardware": {"virtualHost": {"id": {"operation": "not null"}}}}
         mask = "mask[virtualHost[guests[powerState]]]"
         return self.client.call('SoftLayer_Account', 'getHardware', mask=mask, filter=object_filter)
+
+    def authorize_storage(self, vs_id, username_storage):
+        """Authorize File or Block Storage to a Virtual Server.
+
+        :param int vs_id: Virtual server id.
+        :param string username_storage: Storage username.
+
+        :return: bool.
+        """
+        _filter = {"networkStorage": {"username": {"operation": username_storage}}}
+
+        storage_result = self.client.call('Account', 'getNetworkStorage', filter=_filter)
+
+        if len(storage_result) == 0:
+            raise SoftLayerError("The Storage with username: %s was not found, please"
+                                 " enter a valid storage username" % username_storage)
+
+        storage_template = [
+            {
+                "id": storage_result[0]['id'],
+                "username": username_storage
+            }
+        ]
+
+        result = self.client.call('SoftLayer_Virtual_Guest', 'allowAccessToNetworkStorageList',
+                                  storage_template, id=vs_id)
+
+        return result
+
+    def attach_portable_storage(self, vs_id, portable_id):
+        """Attach portal storage to a Virtual Server.
+
+        :param int vs_id: Virtual server id.
+        :param int portable_id: Portal storage id.
+
+        :return: SoftLayer_Provisioning_Version1_Transaction.
+        """
+        disk_id = portable_id
+        result = self.client.call('SoftLayer_Virtual_Guest', 'attachDiskImage',
+                                  disk_id, id=vs_id)
```

Comment:
result = self.client.call('SoftLayer_Virtual_Guest', 'attachDiskImage', portable_id, id=vs_id)
is not better this?

Reply:
I likely would have written `self.client.call('SoftLayer_Virtual_Guest', 'attachDiskImage', portable_id, id=vs_id)` as I think it looks better, taking up less lines.

But there are plenty of cases in the code that would break the call up like this due to line length, so this pattern of breaking up the call into 2 lines after the method is fine too.

Its not something I feel strongly enough about to make you change though.